<launch>

  <node pkg="mcl_3dl" exec="pcl_publisher" output="screen" respawn="false">
    <param from="$(find-pkg-share mcl_3dl)/config/pcl_publisher.yaml" />
    <param name="map_dir" value="$(find-pkg-share perception_3d)/map/map.pcd" />
    <param name="ground_dir" value="$(find-pkg-share perception_3d)/map/ground.pcd" />
  </node>
  
  <!--name argument is not required, because it will overwrite all node name in the exec-->
  <node pkg="perception_3d" exec="perception_3d_ros_node" output="screen" respawn="false">
    <param from="$(find-pkg-share perception_3d)/config/perception_3d_ros.yaml" />
  </node>

  <!--- TF -->
  <node pkg="tf2_ros" exec="static_transform_publisher" name="map2baselink" args="1.0 0 0.0 0.0 0.0 0.0 /map /base_link" />
  <node pkg="tf2_ros" exec="static_transform_publisher" name="baselink2laserlink" args="0.5 0 0.5 0.0 0.0 0.0 /base_link /laser_link" />
  <node pkg="tf2_ros" exec="static_transform_publisher" name="baselink2cameralink" args="0.5 0 0.4 0.0 0.0 0.0 /base_link /camera_link" />
  
  <let name="rviz_config" value="$(find-pkg-share perception_3d)/rviz/zone_demo.rviz"/>
  <node name="rviz2" pkg="rviz2" exec="rviz2" args="-d $(var rviz_config)"/>

</launch>
